python/tests/integration/arcticdb/test_arctic.py

```diff
@@ -92,6 +92,46 @@ def test_get_library(arctic_client):
         _ = ac.get_library("pytest_test_lib", create_if_missing=False, library_options=library_options)
 
 
+def test_create_library_with_invalid_name(arctic_client):
+    ac = arctic_client
+
+    # These should succeed because the names are valid
+    valid_names = ["lib", "lib/with/slash", "lib-with-dash", "lib.with.dot", "lib123"]
+    for lib_name in valid_names:
+        ac.create_library(lib_name)
+
+    # These should fail because the names are invalid
+    invalid_names = [chr(0), "lib>", "lib<", "lib*", "/lib", "lib...lib", "lib"*1000]
+    for lib_name in invalid_names:
+        with pytest.raises(UserInputException):
+            ac.create_library(lib_name)
+
+    # Verify that library list is not corrupted
+    assert set(ac.list_libraries()) == set(valid_names)
+
+
+# TODO: LMDB on Windows raises InternalExceptions because it creates Windows directories with invalid characters.
```

Comment:
Why is this a TODO? There's probably not much we can do about this, short of a complicated name encoding scheme. Shouldn't we just test that LMDB behaves as it currently does?

Reply:
Linked to new issue #1247 instead